### PR TITLE
[fix](local shuffle) Fix hash shuffle local exchanger

### PR DESCRIPTION
**Pull request by Gabriel39**
## Proposed changes

Local Exchange and Exchange in this plan must have same HASH algorithm on multiple BEs
image

### CODE REVIEW

File: be/src/pipeline/pipeline_fragment_context.cpp [hunk 1]
```diff
@@ -692,15 +692,27 @@ Status PipelineFragmentContext::_add_local_exchange_impl(
...
+
+    /**
+     * `bucket_seq_to_instance_idx` is empty if no scan operator is contained in this fragment.
+     * So co-located operators(e.g. Agg, Analytic) should use `HASH_SHUFFLE` instead of `BUCKET_HASH_SHUFFLE`.
+     */
+    const bool should_disable_bucket_shuffle =
+            bucket_seq_to_instance_idx.empty() ||
+            (operator_xs.size() > idx ? operator_xs[idx]->is_shuffled_hash_join()
+                                      : cur_pipe->sink_x()->is_shuffled_hash_join());
     sink.reset(new LocalExchangeSinkOperatorX(
```
Comment (yiguolei) [inline be/src/pipeline/pipeline_fragment_context.cpp:704]: Agg(group a)  ---> join condition (t1.a = t2.a)

scan t2 ----> build

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

**Comment by Gabriel39** [comment]
run buildall

File: be/src/pipeline/exec/analytic_sink_operator.h [hunk 2]
```diff
@@ -82,14 +83,17 @@ class AnalyticSinkOperatorX final : public DataSinkOperatorX<AnalyticSinkLocalSt
...
+            return _is_colocate && _require_bucket_distribution && !_followed_by_shuffled_join
                            ? DataDistribution(ExchangeType::BUCKET_HASH_SHUFFLE, _partition_exprs)
                            : DataDistribution(ExchangeType::HASH_SHUFFLE, _partition_exprs);
         }
         return DataSinkOperatorX<AnalyticSinkLocalState>::required_data_distribution();
     }
 
     bool require_data_distribution() const override { return true; }
+    bool require_shuffled_data_distribution() const override {
+        return !_partition_by_eq_expr_ctxs.empty() && _order_by_eq_expr_ctxs.empty();
```
Comment (yiguolei) [inline be/src/pipeline/exec/analytic_sink_operator.h:95]: OperatorX ---> local shuffle ---> join probe 

OperatorBase
1. require_bucket_distribution
2. followed_by_shuffled_join

**Comment by Gabriel39** [comment]
run buildall

**Review by yiguolei — APPROVED** [review]

**Comment by yiguolei** [comment]
run buildall

**Review by Mryange — APPROVED** [review]

**Pull request merged**